tensorboard/pip_package/build_pip_package.sh

```diff
@@ -135,9 +115,66 @@ pip install -qU wheel 'setuptools>=36.2.0'
 python setup.py bdist_wheel --python-tag py2 >/dev/null
 python setup.py bdist_wheel --python-tag py3 >/dev/null
 
-if [ "$run_smoke_test" = 1 ]; then
-  smoke 2
-  smoke 3
-fi
+smoke() {
+  pymajorversion="$1"
+  if [ -z "${pymajorversion}" ]; then
+    pymajorversion="$(python -c 'import sys; print(sys.version_info[0])')"
+  fi
+  smokepython="python$1"
+  smokevenv="smoke-venv$1"
+  smoketf="$2"
+  set +x
+  printf '\n\n%70s\n' | tr ' ' '='
+  printf "Smoke testing with ${smokepython} and ${smoketf}...\n\n"
+  set -x
+  command -v "${smokepython}" >/dev/null
+  virtualenv -qp "${smokepython}" "${smokevenv}"
+  cd "${smokevenv}"
+  . bin/activate
+  pip install -qU pip
+  pip install -qU "${smoketf}"
+  pip install -qU ../dist/*"py${pymajorversion}"*.whl >/dev/null
+  # Test TensorBoard application
+  [ -x ./bin/tensorboard ]  # Ensure pip package included binary
+  mkfifo pipe
+  tensorboard --port=0 --logdir=smokedir 2>pipe &
+  perl -ne 'print STDERR;/http:.*:(\d+)/ and print $1.v10 and exit 0' <pipe >port
+  curl -fs http://localhost:$(cat port) >index.html
```

Comment:
As you rewrite this, please fix it: `$(cat port)` must be quoted
(personally I’d quote the whole thing). (Here and below.)

Reply:
Ok, but FWIW, all of these lines are only moved, not rewritten.

Reply:
Done.